### PR TITLE
Edge cases for boundary normal vector in coarsen_from_fine

### PR DESCRIPTION
**Pull request by eebasso**
## Summary
Catch edge cases for small nx, ny, nz

## Checklist

The proposed changes:
- [x] fix a bug or incorrect behavior in AMReX
- [ ] add new capabilities to AMReX
- [ ] changes answers in the test suite to more than roundoff level
- [ ] are likely to significantly affect the results of downstream AMReX users
- [ ] include documentation in the code and/or rst files, if appropriate

### CODE REVIEW

File: Src/EB/AMReX_EB2_3D_C.H [hunk 1]
```diff
@@ -481,10 +481,31 @@ int coarsen_from_fine (int i, int j, int k, Box const& bx, int ngrow,
                 +     fbn(ii+1,jj  ,kk+1,2)*fba(ii+1,jj  ,kk+1)
                 +     fbn(ii  ,jj+1,kk+1,2)*fba(ii  ,jj+1,kk+1)
                 +     fbn(ii+1,jj+1,kk+1,2)*fba(ii+1,jj+1,kk+1);
-            Real nfac = 1.0_rt / std::sqrt(nx*nx+ny*ny+nz*nz+1.e-30_rt);
-            cbn(i,j,k,0) = nx*nfac;
-            cbn(i,j,k,1) = ny*nfac;
-            cbn(i,j,k,2) = nz*nfac;
+            Real nxyz = std::sqrt(nx*ny + ny*ny + nz*nz);
```
Comment (WeiqunZhang) [inline Src/EB/AMReX_EB2_3D_C.H]:
```suggestion
            Real nxyz = std::sqrt(nx*nx + ny*ny + nz*nz);
```

**Review by WeiqunZhang — APPROVED** [review]
